### PR TITLE
Feat/add help action

### PR DESCRIPTION
**Pull request by mkondratek**
First version:
image

Second version: 
image

~Tarcza~ Demo window 3.0:
image
 
___

To consider:
- add more and general description (so that the embedded graph table will not be the only component)
- use another column instead of custom annotation to provide the description
- enhance the "example" graph (add subbranch(es) & present STRSes)
- maybe some links here too

### CODE REVIEW

File: frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/cell/BranchOrCommitCellRenderer.java [hunk 1]
```diff
@@ -36,14 +37,13 @@
...
 import com.virtuslab.gitmachete.frontend.graph.api.render.parts.IRenderPart;
-import com.virtuslab.gitmachete.frontend.ui.impl.table.GitMacheteGraphTable;
 
 @UI
 public class BranchOrCommitCellRenderer extends TypeSafeTableCellRenderer<BranchOrCommitCell> {
 
   private final MyComponent myComponent;
 
-  public BranchOrCommitCellRenderer(GitMacheteGraphTable table, IGraphCellPainter painter) {
+  public BranchOrCommitCellRenderer(JBTable table, IGraphCellPainter painter) {
```
Comment (PawelLipski) [inline frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/cell/BranchOrCommitCellRenderer.java:46]: :+1:

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/HelpAction.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      super(/* canBeParent */ false);
+      init();
+      setTitle("Git Machete Help");
+    }
+
+    @Override
+    protected JComponent createCenterPanel() {
+      var panel = JBUI.Panels.simplePanel(0, 2);
+
+      var iRepositoryGraphFactory = RuntimeBinding.instantiateSoleImplementingClass(IRepositoryGraphFactory.class);
```
Comment (PawelLipski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/HelpAction.java:50]: drop `i` ;)

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/HelpAction.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+import com.virtuslab.binding.RuntimeBinding;
+import com.virtuslab.gitmachete.frontend.graph.api.coloring.GraphItemColorToJBColorMapper;
+import com.virtuslab.gitmachete.frontend.graph.api.paint.IGraphCellPainterFactory;
+import com.virtuslab.gitmachete.frontend.graph.api.repository.IRepositoryGraphFactory;
+import com.virtuslab.gitmachete.frontend.ui.impl.cell.BranchOrCommitCell;
+import com.virtuslab.gitmachete.frontend.ui.impl.cell.BranchOrCommitCellRenderer;
+import com.virtuslab.gitmachete.frontend.ui.impl.table.GraphTableModel;
+
+/**
+ * Expects DataKeys:
```
Comment (PawelLipski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/HelpAction.java:26]:
```suggestion
 * Expects DataKeys: <none>
```

File: frontendActions/build.gradle [hunk 4]
```diff
@@ -7,6 +7,8 @@ dependencies {
   implementation project(':branchLayoutApi')
   implementation project(':frontendActionIds')
   implementation project(':frontendDataKeys')
+  implementation project(':frontendGraphApi')
+  implementation project(':frontendUiTableImpl')
```
Comment (PawelLipski) [inline frontendActions/build.gradle:11]: :thinking: maybe expose more stuff to `frontendUiApi` instead? ofc not if that's gonna turn out too much engineering

File: frontendUiTableDemo/src/main/java/com/virtuslab/gitmachete/frontend/ui/demo/DemoGitMacheteRepository.java [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+public class DemoGitMacheteRepository implements IGitMacheteRepository {
+
+  private final List<IGitMacheteRootBranch> roots;
+
+  public DemoGitMacheteRepository() {
+    var nullPointedCommit = new Commit("");
+    var fp = new FpCommit("Fork point commit");
+    NonRoot[] nonRoots = {
+        new NonRoot(/* name */ "allow-ownership-link",
+            /* customAnnotation */ "#Gray edge: branch is merged to its parent branch",
```
Comment (PawelLipski) [inline frontendUiTableDemo/src/main/java/com/virtuslab/gitmachete/frontend/ui/demo/DemoGitMacheteRepository.java:38]: Nit: I'd add a space after hash `# ` everywhere

File: frontendUiTableDemo/src/main/java/com/virtuslab/gitmachete/frontend/ui/demo/DemoGitMacheteRepository.java [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+        new NonRoot(/* name */ "build-chain",
+            /* customAnnotation */ "#Green edge: branch is in sync to its parent branch",
+            nullPointedCommit,
+            /* fork point */ null,
+            /* downstreamBranches */ List.empty(),
+            /* commits */ List.of(new Commit("Second commit of build-chain"),
+                new Commit("First commit of build-chain")),
+            SyncToParentStatus.InSync),
+        new NonRoot(/* name */ "call-ws",
+            /* customAnnotation */ "#Yellow edge: Branch is in sync to its parent branch but the fork point is not equal to parent branch",
```
Comment (PawelLipski) [inline frontendUiTableDemo/src/main/java/com/virtuslab/gitmachete/frontend/ui/demo/DemoGitMacheteRepository.java:53]: To increase readability:
```suggestion
            /* customAnnotation */ "# Yellow edge: Branch is in sync to its parent branch, but the fork point is NOT equal to parent branch",
```

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/HelpAction.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+      init();
+      setTitle("Git Machete Help");
+    }
+
+    @Override
+    protected JComponent createCenterPanel() {
+      var panel = JBUI.Panels.simplePanel(0, 2);
+      panel.addToCenter(ScrollPaneFactory.createScrollPane(DemoGraphTable.INSTANCE));
+      panel.setPreferredSize(new JBDimension(CENTER_PANEL_WIDTH, CENTER_PANEL_HEIGHT));
+      return panel;
```
Comment (micpiotrowski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/HelpAction.java:39]: If it's easy doable please remove one of buttons from this panel (preferably `Cancel` button)

Comment (mkondratek) [reply frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/HelpAction.java:48]: good point, done :tada:

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/HelpAction.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    protected Action[] createActions() {
+      Action helpAction = getHelpAction();
+      return helpAction == myHelpAction && getHelpId() == null
+          ? new Action[]{getOKAction()}
+          : new Action[]{getOKAction(), helpAction};
+    }
+
+    @Override
+    protected JComponent createCenterPanel() {
+      var panel = JBUI.Panels.simplePanel(0, 2);
```
Comment (PawelLipski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/HelpAction.java:45]: Pls name the params (can be on another PR)
